### PR TITLE
Compare

### PR DESCRIPTION
**Pull request by alirezazolanvari**
#### What does this implement/fix? Explain your changes.
Enhance the performance of `compare` by adding four more overall benchmarks and editing the default overall benchmarks' weights.  

#### Any other comments?
@sepandhaghighi Please modify `Document.ipynb` and `README.md` respectively.

#### References
[1]	 K. Villela, A. Silva, T. Vale, and E. S. de Almeida, "A survey on software variability management approaches," in _Proceedings of the 18th International Software Product Line Conference-Volume 1_, 2014, pp. 147-156. 
[2]	J. R. Saura, A. Reyes-Menendez, and P. Palos-Sanchez, "Are black Friday deals worth it? Mining Twitter users’ sentiment and behavior response," _Journal of Open Innovation: Technology, Market, and Complexity_, vol. 5, no. 3, p. 58, 2019.
[3]	P. Schubert and U. Leimstoll, "Importance and use of information technology in small and medium‐sized companies," _Electronic Markets_, vol. 17, no. 1, pp. 38-55, 2007.

### CODE REVIEW

File: Test/compare_test.py [hunk 1]
```diff
@@ -66,19 +66,19 @@
 >>> print(cp3)
 Best : cm2
 <BLANKLINE>
-Rank  Name   Class-Score    Overall-Score
-1     cm2    0.70556        0.96667
-2     cm1    0.55           0.74722
+Rank  Name   Class-Score       Overall-Score
+1     cm2    0.70556           0.92381
+2     cm1    0.55              0.75238
```
Comment (sadrasabouri) [inline Test/compare_test.py:71]: What is this new alignments for?

File: pycm/pycm_param.py [hunk 2]
```diff
@@ -368,6 +368,38 @@
...
 
 
+SOA7_SCORE = {
+    "Perfect": 6,
+    "Very Strong": 5,
+    "Strong": 4,
+    "Moderate": 3,
+    "Weak": 2,
+    "Very Weak": 1,
+    "None": "None"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:378]: I recommend the bellow format:
```python
    "None": "None",
}
```

File: pycm/pycm_param.py [hunk 3]
```diff
@@ -368,6 +368,38 @@
...
+
+SOA8_SCORE = {
+    "Perfect": 6,
+    "Very Strong": 5,
+    "Strong": 4,
+    "Moderate": 3,
+    "Weak": 2,
+    "Very Weak": 1,
+    "None": "None"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:387]:
```python
    "None": "None",
}
```

File: pycm/pycm_param.py [hunk 4]
```diff
@@ -368,6 +368,38 @@
...
+
+SOA9_SCORE = {
+    "High": 3,
+    "Tentative": 2,
+    "Low": 1,
+    "None": "None"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:393]:
```python
    "None": "None",
}
```

File: pycm/pycm_param.py [hunk 5]
```diff
@@ -368,6 +368,38 @@
...
+
+SOA10_SCORE = {
+    "Strong": 4,
+    "Medium": 3,
+    "Weak": 2,
+    "Not Appreciable": 1,
+    "None": "None"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:400]:
```python
    "None": "None",
}
```

File: pycm/pycm_param.py [hunk 6]
```diff
@@ -384,17 +416,26 @@
...
     "SOA2": "SOA2(Fleiss)",
     "SOA3": "SOA3(Altman)",
     "SOA4": "SOA4(Cicchetti)",
     "SOA5": "SOA5(Cramer)",
-    "SOA6": "SOA6(Matthews)"}
+    "SOA6": "SOA6(Matthews)",
+    "SOA7": "SOA7(Lambda A)",
+    "SOA8": "SOA8(Lambda B)",
+    "SOA9": "SOA9(Krippendorff Alpha)",
+    "SOA10": "SOA10(Pearson C)"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:438]:
```python
   "SOA10": "SOA10(Pearson C)",
}
```

File: pycm/pycm_param.py [hunk 7]
```diff
@@ -601,7 +642,11 @@
     "Bangdiwala B": "Bangdiwala's-B",
     "Krippendorff Alpha": "Krippendorff's-alpha",
     "HD": "HD-(Hamming-distance)",
-    "BB": "BB-(Braun-Blanquet-similarity)"}
+    "BB": "BB-(Braun-Blanquet-similarity)",
+    "SOA7(Lambda A)": "Goodman-&-Kruskal's-lambda-A-interpretation",
+    "SOA8(Lambda B)": "Goodman-&-Kruskal's-lambda-B-interpretation",
+    "SOA9(Krippendorff Alpha)": "Krippendorff's-alpha-interpretation",
+    "SOA10(Pearson C)": "Pearson's-C-interpretation"}
```
Comment (sadrasabouri) [inline pycm/pycm_param.py:649]:
```python
    "SOA10(Pearson C)": "Pearson's-C-interpretation"}
}
```

File: pycm/pycm_util.py [hunk 8]
```diff
@@ -716,3 +717,16 @@ def thresholds_calc(probs):
...
+    """
+    Sort a list of strings first alphabetically and then numerically.
+    
+    :param input_list: input list of strings
+    :type input_list: iterable
+    :return: a sorted list of strings
+    """
+    return sorted(input_list, key=lambda x: [(x, False, False) if not re.findall(r'\d+', x)
+                                             else (x[:re.search(r'\d+', x).start()], int(re.findall(r'\d+', x)[0]),
+                                                   x[re.search(r'\d+', x).end():])])
```
Comment (sadrasabouri) [inline pycm/pycm_util.py:732]: We may need an `autopep8.sh` here. I also suggest the below snippet:
```python
sort_by = lambda x: [(x, False, False) if not re.findall(r'\d+', x) \
                else (x[:re.search(r'\d+', x).start()],
                      int(re.findall(r'\d+', x)[0]),
                      x[re.search(r'\d+', x).end():])])
return sorted(input_list, key=sort_by)
```

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Minor changes needed.

**Review by sadrasabouri — APPROVED** [review]

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🚀

**Comment by sepandhaghighi** [comment]
@sadrasabouri @alirezazolanvari Please re-review this PR 💯 
If everything is okay, **Squash/Merge** it after approval 🚀

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**